sync/class.jetpack-sync-item.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+
+class Jetpack_Sync_Item {
+
+	private $object; // The object we are syncing, eg a post
+	private $terms; // Terms related to the object, eg categories and tags
+	private $state; // The state of the object, eg `is_just_published`
+	private $trigger; // The action that triggers the sync operation, eg `save_post`
+
+	function __construct( $trigger, $object = null ) {
+		$this->trigger = $trigger;
+		if ( ! is_null( $object ) ) {
+			$this->set_object( $object );
+		}
+	}
+
+	function add_terms( Jetpack_Sync_Item $terms ) {
+		$this->terms[] = $terms;
```

Comment:
Will we always have one batch of terms to add to a sync object? Would there be any reason to array_merge the array of terms that are passed as an argument with the existing array of terms?

Reply:
This is consistent with how we've set object terms in the past, but i agree that it should be more clear. we can address that in an other PR